core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -2076,4 +2076,17 @@ public static boolean isAuditEnabled() {
   public static void setAuditEnabled(boolean enabled) {
     getInstance().addProperty(CarbonCommonConstants.CARBON_ENABLE_AUDIT, String.valueOf(enabled));
   }
+
+  public boolean isSIRepairEnabledInLoad() {
+    String configuredValue = getSessionPropertyValue(CarbonCommonConstants.CARBON_LOAD_SI_REPAIR);
+    if (configuredValue == null) {
+      // if not set in session properties then check carbon.properties for the same
+      configuredValue = getProperty(CarbonCommonConstants.CARBON_LOAD_SI_REPAIR);
```

Comment:
`getProperty(String key)` will internally call `getSessionPropertyValue(key)` to get the session level configured value if available otherwise returns carbon level vaue. so remove the explicit call to `getSessionPropertyValue()`
Also suggest you to use `getProperty(String key, String defaultValue)` instead of `getProperty(String key)`. We can avoid these null checks